File: src/main/kotlin/h/MainNormal.java
package h;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class MainNormal {
    public static void main(String[] args) {
        Map<Integer, String> numberMap = IntStream.range(0, 30 + 1)
                .filter(value -> value % 2 == 0)
                .map(value -> value * 10)
                .boxed()
                .collect(Collectors.toMap(
                        Function.identity(),
                        value -> Integer.toString(value, 16),
                        (a, b) -> {throw new IllegalStateException();},
                        LinkedHashMap::new));

        System.out.printf("[%s] %s%n", numberMap.getClass().getName(), numberMap);

        // an alternative way

        Map<Integer, String> numberMapClassical = new LinkedHashMap<>();
        for (int i = 0; i <= 30; i++) {
            if (i % 2 == 0) {
                int value = i * 10;
                numberMapClassical.put(value, Integer.toString(value, 16));
            }
        }

        System.out.println("\nClassic");
        System.out.printf("[%s] %s%n", numberMapClassical.getClass().getName(), numberMapClassical);
    }
}

File: src/main/kotlin/h/MainNormal.kt
package h

import java.lang.IllegalStateException
import java.util.function.Function
import java.util.stream.Collectors
import java.util.stream.IntStream

fun main() {
    val numberMap: Map<Int, String> =
        (0..30).asSequence()
            .filter { it % 2 == 0 }
            .map { it * 10 }
            .associateWith { it.toString(16) }

    println("[${numberMap.javaClass.name}] $numberMap")

    // an alternative way

    val numberMapClassical = LinkedHashMap<Int, String>()
    for (i in 0..30) {
        if (i % 2 == 0) {
            val value = i * 10
            numberMapClassical[value] = value.toString(16)
        }
    }

    println("\nClassic")
    println("[${numberMapClassical.javaClass.name}] $numberMapClassical")

    // java style

    val numberMapJava = IntStream.range(0, 30 + 1)
        .filter { it % 2 == 0 }
        .map { it * 10 }
        .boxed()
        .collect(Collectors.toMap(
            Function.identity(),
            { it.toString(16) },
            { _, _ -> throw IllegalStateException() },
            ::LinkedHashMap
        ))

    println("\nJava")
    println("[${numberMapJava.javaClass.name}] $numberMapJava")
}